File: backend/dotnet/src/Core/Workforce.AppCore/Domain/Dashboard/DashboardSummary.cs
namespace Workforce.AppCore.Domain.Dashboard;

public sealed class DashboardSummary
{
    public string Id { get; set; } = string.Empty;
    public DateTimeOffset GeneratedAt { get; set; }
    public List<DepartmentCount> HeadcountByDepartment { get; set; } = [];
    public int ActiveProjectsCount { get; set; }
    public List<StatusCount> TasksByStatus { get; set; } = [];
    public List<LeaveStat> LeaveStats { get; set; } = [];

    public sealed class DepartmentCount
    {
        public string Department { get; set; } = string.Empty;
        public int Count { get; set; }
    }

    public sealed class StatusCount
    {
        public string Status { get; set; } = string.Empty;
        public int Count { get; set; }
    }

    public sealed class LeaveStat
    {
        public string Type { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public int Count { get; set; }
    }
}

File: backend/node/dashboard-worker/src/domain/DashboardSummary.ts
export type HeadcountByDepartment = {
  department: string;
  count: number;
};

export type TasksByStatus = {
  status: string;
  count: number;
};

export type LeaveStat = {
  type: string;
  status: string;
  count: number;
};

export type DashboardSummary = {
  summaryKey: string;
  generatedAt: Date;
  headcountByDepartment: HeadcountByDepartment[];
  activeProjectsCount: number;
  tasksByStatus: TasksByStatus[];
  leaveStats: LeaveStat[];
};
